### PR TITLE
feat(document-numbering): add support for new document numbering fields

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Currently Lago does not provide solution for updating document numbers. Also, `per_customer` is the only approach in creating incremental numbers. With this feature, it will be allowed to update one part of document number. Also, it will be possible to choose between `per_customer` and `per_organization` numbering 

## Description

This PR adds API and graphql support for new document numbering fields.
It also sets by default `document_number_prefix` as combination of organization name and id.

### CODE REVIEW

File: app/graphql/types/organization_type.rb [hunk 1]
```diff
@@ -27,6 +27,9 @@ class OrganizationType < Types::BaseObject
 
     field :timezone, Types::TimezoneEnum, null: true
 
+    field :document_numbering, Types::Organizations::DocumentNumberingEnum, null: false
+    field :document_number_prefix, String, null: false
```
Comment (jdenquin) [inline app/graphql/types/organization_type.rb:31]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `document_number_prefix` should appear before `document_numbering`.

File: app/graphql/types/organizations/document_numbering_enum.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Types
+  module Organizations
+    class DocumentNumberingEnum < Types::BaseEnum
```
Comment (jdenquin) [inline app/graphql/types/organizations/document_numbering_enum.rb:5]: GraphQL/GraphqlName: graphql_name should be specified only for overrides.

File: app/graphql/types/organizations/update_organization_input.rb [hunk 3]
```diff
@@ -24,6 +24,9 @@ class UpdateOrganizationInput < BaseInputObject
 
       argument :timezone, Types::TimezoneEnum, required: false
 
+      argument :document_numbering, Types::Organizations::DocumentNumberingEnum, required: false
+      argument :document_number_prefix, String, required: false
```
Comment (jdenquin) [inline app/graphql/types/organizations/update_organization_input.rb:28]: GraphQL/OrderedArguments: Arguments should be sorted in an alphabetical order within their section. Field `document_number_prefix` should appear before `document_numbering`.

File: app/models/organization.rb [hunk 4]
```diff
@@ -94,6 +103,10 @@ def generate_api_key
     self.api_key = SecureRandom.uuid
   end
 
+  def generate_document_number_prefix
+    self.update!(document_number_prefix: "#{name.first(3).upcase}-#{id.last(4).upcase}")
```
Comment (jdenquin) [inline app/models/organization.rb:107]: Style/RedundantSelf: Redundant `self` detected. (https://rubystyle.guide#no-self-unless-required)

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**